packages/google-vertex/src/edge/google-vertex-provider-edge.ts

```diff
@@ -0,0 +1,26 @@
+import { generateAuthToken } from './google-vertex-auth-edge';
+import {
+  createVertex as createVertexOriginal,
+  GoogleVertexProvider,
+  GoogleVertexProviderSettings,
+} from '../google-vertex-provider';
+
+export type { GoogleVertexProviderSettings, GoogleVertexProvider };
+
+export function createVertex(
+  options: GoogleVertexProviderSettings = {},
+): GoogleVertexProvider {
+  return createVertexOriginal({
+    ...options,
+    headers:
+      options.headers ??
+      (async () => ({
+        Authorization: `Bearer ${await generateAuthToken()}`,
```

Comment:
Do you want to pass through some of the options into `generateAuthToken`? or leave for later?

Reply:
Yes, thanks. Added `googleCredentials?`.